Update lambda-calculus-bonus-exercises-answers.md

skilgal opened the pull request:
Fix the typo on task 4

skilgal committed:
Update lambda-calculus-bonus-exercises-answers.md

Fix the typo on task 4

k-bx commented:
There is indeed an error, but not where you've fixed it :) Here's the correct fix:

![IMG_0240](https://user-images.githubusercontent.com/107968/67100169-e12b2d80-f1c7-11e9-87d2-9130848f4e8f.JPG)

skilgal commented:
Actually I can't see the difference.

My substitution is `[y -> y1]` and your's `[y -> w]` . 
but the other part looks the same. 
There is some difference in the braces, but according to the [CR..smth that](https://www.youtube.com/watch?v=KoIdCHDbpMI) it shouldn't make the difference.

`λw1. (λz. (λc. y)z)w <=> λy1. ((λz. (λc. y)z)y1)`

for me, it looks like 
`λx. x y <=> λx. (x y)`

Correct me if I'm wrong

skilgal committed:
Update lambda-calculus-bonus-exercises-answers.md

add lambda transformation from the beginning
changed an alias to new letter without a number

skilgal commented:
Okay, maybe found what you meant. 
The lambda transformation should be done as first since on my second line we get a collision

k-bx reviewed (approved):
I would explicitly add alpha-renaming as a step before doing the beta-reduction, but this looks correct now.
